app/main/views/index.py

```diff
@@ -32,6 +37,13 @@ def download_document_download(service_id, document_id):
     if not key:
         abort(404)
 
+    user_id = session.get('user_id')
+
+    # the user hasn't clicked through the landing page - redirect them back there
+    if current_user.get_id() != user_id:
```

Comment:
Right now this basically checks whether the user has hit the same app instance they've got the cookie from.